collector/terra/fcd/fcdepo.go

```diff
@@ -0,0 +1,144 @@
+package fcd
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dezswap/cosmwasm-etl/pkg/db/schemas"
+	"github.com/dezswap/cosmwasm-etl/pkg/terra/fcd"
+	"github.com/pkg/errors"
+)
+
+type columbus4FcdRepo struct {
+	client fcd.Fcd
+	col4Mapper
+}
+
+func NewFcdRepo(client fcd.Fcd) fcdRepo {
+	return &columbus4FcdRepo{client, &col4RepoMapper{}}
+}
+
+// HasMoreTx implements fcdRepo.
+func (rp *columbus4FcdRepo) HasMoreTx(addr string, collectedOffset uint32) (bool, error) {
+	limit := fcd.FCD_TXS_MAX_LIMIT
+	offset := int(collectedOffset)
+	txsRes, err := rp.client.TxsOf(addr, fcd.FcdTxsReqQuery{Limit: &limit, Offset: &offset})
+	if err != nil {
+		return false, errors.Wrap(err, "columbus4SrcRepo.TxsOf")
+	}
+	return len(txsRes.Txs) != 0, nil
+}
+
+// / TxsOf retrieves transactions from the FCD server and assumes that transactions greater than the offset have already been stored.
+//
+// @collectedOffset the offset of the last transaction collected from the FCD server.
+//
+// @untilHeight desired block until which transactions are intended to be collected.
+// it is used to find the starting offset (the FCD server returns in reverse order of height)
+func (rp *columbus4FcdRepo) TxsOf(addr string, collectedOffset, untilHeight, maxLen uint32) ([]schemas.FcdTxLog, error) {
+	startOffset := collectedOffset
+	var err error
+	if collectedOffset == 0 {
+		startOffset, err = rp.findStartOffset(addr, untilHeight)
+		if err != nil {
+			return nil, errors.Wrap(err, "columbus4SrcRepo.TxsOf")
+		}
+	}
+
+	errLimit := 3
+	limit := fcd.FCD_TXS_MAX_LIMIT
+	offset := int(startOffset)
+	txs := []schemas.FcdTxLog{}
+	for ; ; offset = int(txs[len(txs)-1].FcdOffset) {
+		txsRes, err := rp.client.TxsOf(addr, fcd.FcdTxsReqQuery{Limit: &limit, Offset: &offset})
+		if err != nil {
+			if strings.Contains(err.Error(), fcd.STATUS_SERVER_ERROR.Error()) && 0 < errLimit {
+				errLimit--
+				time.Sleep(10 * time.Second)
+				fmt.Print(err.Error())
```

Comment:
please delete this line

Reply:
https://github.com/dezswap/cosmwasm-etl/pull/15/commits/c07da70eaeb89d08fa1de79b63e592ff1769b041